routers/repo/download.go

```diff
@@ -40,6 +42,12 @@ func ServeData(ctx *context.Context, name string, reader io.Reader) error {
 		ctx.Resp.Header().Set("Content-Disposition", fmt.Sprintf(`attachment; filename="%s"`, name))
 	}
 
+	if true == ctx.Data["IsRawApi"] {
```

Comment:
```suggestion
	if ctx.Data["IsRawApi"] {
```

Reply:
If I change the code like this, compilation fails with

`non-bool ctx.Context.Data["IsRawApi"] (type interface {}) used as if condition`

Reply:
This is with go v1.13.4.

As far as I can see, the following approaches are used in gitea to tackle corresponding situations:

- routers/api/v1/api.go:          `if true == ctx.Data["IsApiToken"] {`
- routers/repo/issue.go:          `if !ctx.Data["IsStopwatchRunning"].(bool) {`
- routers/user/home_test.go:      `assert.EqualValues(t, true, ctx.Data["IsShowClosed"])`

I prefer the first but am very open to further suggestions!

Reply:
@danielappelt I agree, I think your original is best.

Reply:
imho than switching order would be better: `ctx.Data["IsRawApi"] == true`

Reply:
The two most likely reasons to have the constant in front are

- Avoiding `=` vs `==` errors. This should not be a concern though as the compiler would fail with an error anyway.
- Having the expression stand out from the rest which would maybe make people think why an explicit comparison is needed in this case.

For readability, I also prefer your suggestion. So, I just updated the code accordingly.

Reply:
We've received the suggestion about yoda conditionals before, and they do have their benefits (for the reasons you've listed). If we were to update them then changing them all at once in their own PR so that the codebase is consistent (although a discussion would probably be needed first)